controllers/quickstarts_test.go

```diff
@@ -115,3 +116,116 @@ func getActualQuickStarts(t *testing.T, cases []struct {
 	}
 	return allActualQuickStarts
 }
+
+func TestDeleteQuickStarts(t *testing.T) {
+	fss1 := generateFakeStorageSystem()
+	fss2 := generateFakeStorageSystem()
+
+	fss3 := generateFakeStorageSystem()
+	fss4 := generateFakeStorageSystem()
+
+	fss5 := generateFakeStorageSystem()
+
+	testCases := []struct {
+		label                string
+		createStorageSystems []odfv1alpha1.StorageSystem
+		deleteStorageSystems []odfv1alpha1.StorageSystem
+		expectDeleted        bool
+	}{
+		{
+			label:                "having two storage systems but only deleting one does not delete quickstarts",
+			createStorageSystems: []odfv1alpha1.StorageSystem{fss1, fss2},
+			deleteStorageSystems: []odfv1alpha1.StorageSystem{fss1},
+			expectDeleted:        false,
+		},
+		{
+			label:                "having two storage systems and deleting both deletes the quickstarts",
+			createStorageSystems: []odfv1alpha1.StorageSystem{fss3, fss4},
+			deleteStorageSystems: []odfv1alpha1.StorageSystem{fss3, fss4},
+			expectDeleted:        true,
+		},
+
+		{
+			label:                "having one storage system and deleting it deletes the quickstarts",
+			createStorageSystems: []odfv1alpha1.StorageSystem{fss5},
+			deleteStorageSystems: []odfv1alpha1.StorageSystem{fss5},
+			expectDeleted:        true,
+		},
+	}
+
+	fakeReconciler, fakeStorageSystem := GetFakeStorageSystemReconciler()
+	err := operatorv1alpha1.AddToScheme(fakeReconciler.Scheme)
+	assert.NoError(t, err)
+
+	// Cleanup of default StorageSystem for clean tests
+	err = fakeReconciler.Client.Delete(context.TODO(), fakeStorageSystem)
+	assert.NoError(t, err)
```

Comment:
So these should come into the for loop itself. With every test it should be a new reconcile and objects so that prev TC does not effect the next one. Moreover, you just need to create only 2 storageSystems which you can make use of in all test cases.

Reply:
fixed the tests. I have moved the reconciler into the for loop and removed the extra ss